neurovault/apps/statmaps/views.py

```diff
@@ -650,6 +658,63 @@ def atlas_query_voxel(request):
         return JSONResponse('error: one or more coordinates are out of range', status=400)
     return JSONResponse(data)
 
+# Compare Two Images
+def compare_images(request,pk1,pk2):
+    image1 = get_image(pk1,None,request)
+    image2 = get_image(pk2,None,request)
+    
+    # Name the data file based on the new volumes
+    path, name1, ext = split_filename(image1.file.url)
+    path, name2, ext = split_filename(image2.file.url)
+
+    # For now, manually choose an atlas - this can be user choice
+    atlas_file = os.path.abspath(os.path.join(os.path.dirname( neurovault.settings.BASE_DIR ), '../..','image_data/atlas/MNI-maxprob-thr25-2mm.nii'))
+    atlas_xml = os.path.abspath(os.path.join(os.path.dirname( neurovault.settings.BASE_DIR ), '../..','image_data/atlas/MNI.xml'))
+    atlas = pybrainatlas.atlas(atlas_xml,atlas_file) # Default slices are "coronal","axial","sagittal"
+
+    # Create custom image names for the visualization
+    custom = {"image 1":"%s:%s" %(image1.name,image1.map_type),"image 2": "%s:%s" %(image2.name,image2.map_type)}
+
+    html_snippet,data_table = compare.scatterplot_compare(image1=image1.file.path,image2=image2.file.path,software="FREESURFER",voxdim=[8,8,8],atlas=atlas,custom=custom,corr="pearson")
+
+    html = [h.strip("\n") for h in html_snippet]
+    context = {'html': html}
+    return render(request, 'statmaps/compare_images.html', context)
+
+# Return search interface for one image vs rest
+def compare_search(request,pk1):
+    image1 = get_image(pk1,None,request)
+    # This df should ONLY contain public images! see tasks.py for generation
+    corr_df = pandas.read_pickle(os.path.abspath(os.path.join(os.path.dirname( neurovault.settings.BASE_DIR ), '../..', 'image_data/matrices/pearson_corr.pkl')))
+    public_images = Image.objects.filter(collection__private=False,id__in=corr_df.columns)
+
+    # Get all image png paths
+    image_paths = [image.file.path for image in public_images]
+    png_img_names = ["glass_brain_%s.png" % image.pk for image in public_images]  
+    png_img_paths = [ os.path.join(os.path.split(image_paths[i])[0],png_img_names[i]) for i in range(0,len(image_paths))]
+    png_img_paths = [imagepath.replace("opt/image_data","media") for imagepath in png_img_paths]  
```

Comment:
Oh, I think I fixed this one, I'm not sure why it is the old one up here! Take a look here: https://github.com/NeuroVault/NeuroVault/pull/111/files#diff-7129b607d02d4a1b46380abd3fd7d834R692 it was the same suggestion by Gabriel to use the image.file.url instead. I'm not totally sure about how this works, but (I think) the media/images path isn't actually a folder but just a URL path that goes to wherever image stuffs is: https://github.com/vsoch/NeuroVault/blob/master/neurovault/apps/statmaps/urls.py#L105